lib/bolt/inventory/group2.rb

```diff
@@ -60,9 +66,41 @@ def initialize(data)
           end
         end
 
-        @groups = groups.map { |g| Group2.new(g) }
+        @groups = groups.map { |g| Group2.new(g, plugins) }
       end
 
+      def validate_config_plugin(data, key, group_name = nil)
+        if data.is_a?(Hash) && data.include?('_plugin')
+          if group_name
+            raise ValidationError.new("Cannot set target #{key.inspect} with config plugin", group_name)
```

Comment:
I think we should generalize these messages to something like "No plugin type  available for setting #{key.inspect} on a target"